crates/bevy_pbr/src/render/mesh.rs

```diff
@@ -513,22 +547,62 @@ impl FromWorld for MeshPipeline {
                 entries.extend_from_slice(&prepass::get_bind_group_layout_entries(
                     [17, 18, 19, 20],
                     multisampled,
+                    depth_prepass,
+                    normal_prepass,
+                    motion_vector_prepass,
+                    deferred_prepass,
                 ));
             }
 
             entries
         }
 
-        let view_layout = render_device.create_bind_group_layout(&BindGroupLayoutDescriptor {
-            label: Some("mesh_view_layout"),
-            entries: &layout_entries(clustered_forward_buffer_binding_type, false),
-        });
+        let mut i = 0;
+        let view_layouts = [(); 32].map(|_| {
+            let multisampled = i & 1_usize != 0;
```

Comment:
The _usize seems unnecessary. And relying on operator precedence for bitwise operations can be risky. As such I basically always write this like `(i & (1 << 0)) != 0` and then `1 << 1` etc. I know people know their powers of 2 but the compiler will evaluate the compile-time constant, and it makes it clear where the bit is.

Reply:
Have you considered using the `bitflags` crate? It's already a dependency of `bevy_pbr` and will allow us to keep this up to date more easily. For example, you could do:

```rust
bitflags! {
    struct MeshLayoutKey: u32 { ... }
}
...
const MESH_LAYOUT_COUNT: usize = MeshLayoutKey::all().bits() as usize + 1;
```

And then instead of hardcoding 32 in there, you could just use `MESH_LAYOUT_COUNT` and the bitflags will automatically compute the correct number, through the magic of `const fn`.

Reply:
I added the `_usize` because clippy was complaining, I didn't consider parenthesis would help. I agree it doesn't look great with the ambiguous visual precedence for relatively uncommon operators.

I'll refactor this as bitflags and use this const fn evaluation trick to make it dynamic, this looks much nicer.